File: Section-5/OO-Kotlin-Classes-Functions-Inheritance/src/com/mx/java/code/_46/ComparisonObjects/SomeClass.java
package com.mx.java.code._46.ComparisonObjects;

public class SomeClass {

    private static int privateVar = 6;

    public static void main(String[] args) {
        new SomeOtherClass().someOtherMethod();
    }

    public static void accesPrivateVar() {
        System.out.println("Im accesing privateVar " + privateVar);
    }
}

File: Section-5/OO-Kotlin-Classes-Functions-Inheritance/src/com/mx/kotlin/_46/ComparisonObjects/SomeClass.kt
package com.mx.kotlin._46.ComparisonObjects

class SomeClass private constructor(val someString: String){
    private val privateVar = 6

    fun accesPrivateVar() {
        System.out.println("Im accesing privateVar $privateVar")
    }

    //Only one companion object is allowed per class
    companion object SomeCompanionObject{
        private  var privateVar = 6

        fun accesPrivateVar() = "Im accesing privateVar $privateVar"

        fun justAssing(str: String) = SomeClass(str)
        fun upperOrLowerCase(str: String, lowerCase: Boolean) : SomeClass{
            return if (lowerCase) {
                SomeClass(str.toLowerCase())
            }
            else {
                SomeClass(str.toLowerCase())
            }
        }

    }
}